extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/BuildTimeContentProcessor.java

```diff
@@ -444,6 +465,108 @@ private void addVersionInfoBuildTimeData(BuildTimeConstBuildItem internalBuildTi
+        ideInfo.put("disable", disable);
+        if (effectiveIdeBuildItem.isPresent()) {
+            EffectiveIdeBuildItem eibi = effectiveIdeBuildItem.get();
+            if (!disable) {
+                // Add IDE info
+                Ide ide = eibi.getIde();
+                ideInfo.put("ideName", ide.name());
+                ideInfo.put("idePackages", getAllUserPackages());
+            }
+        }
+        internalBuildTimeData.addBuildTimeData("ideInfo", ideInfo);
+    }
+
+    private List<String> getAllUserPackages() {
+        List<Path> sourcesDir = DevConsoleManager.getHotReplacementContext().getSourcesDir();
+        List<String> packages = new ArrayList<>();
+
+        for (Path sourcePaths : sourcesDir) {
+            packages.addAll(sourcePackagesForRoot(sourcePaths));
+        }
+        return packages;
+    }
+
+    /**
+     * Return the most general packages used in the application
+     * <p>
+     * TODO: this likely covers almost all typical use cases, but probably needs some tweaks for extreme corner cases
+     */
+    private List<String> sourcePackagesForRoot(Path langPath) {
+        if (!Files.exists(langPath)) {
+            return Collections.emptyList();
+        }
+        File[] rootFiles = langPath.toFile().listFiles();
+        List<Path> rootPackages = new ArrayList<>(1);
+        if (rootFiles != null) {
+            for (File rootFile : rootFiles) {
+                if (rootFile.isDirectory()) {
+                    rootPackages.add(rootFile.toPath());
+                }
+            }
+        }
+        if (rootPackages.isEmpty()) {
+            return List.of("");
+        }
+        List<String> result = new ArrayList<>(rootPackages.size());
+        for (Path rootPackage : rootPackages) {
+            List<String> paths = new ArrayList<>();
+            SimpleFileVisitor<Path> simpleFileVisitor = new DetectPackageFileVisitor(paths);
+            try {
+                Files.walkFileTree(rootPackage, simpleFileVisitor);
+                if (paths.isEmpty()) {
+                    continue;
+                }
+                String commonPath = commonPath(paths);
+                String rootPackageStr = commonPath.replace(langPath.toAbsolutePath().toString(), "")
+                        .replace(File.separator, ".");
+                if (rootPackageStr.startsWith(".")) {
+                    rootPackageStr = rootPackageStr.substring(1);
+                }
+                if (rootPackageStr.endsWith(".")) {
+                    rootPackageStr = rootPackageStr.substring(0, rootPackageStr.length() - 1);
+                }
+                result.add(rootPackageStr);
+            } catch (IOException e) {
+                log.debug("Unable to determine the sources directories", e);
+                // just ignore it as it's not critical for the DevUI functionality
+            }
+        }
+        return result;
```

Comment:
I wonder if there is a better way to get this information as I'm wondering how much it will add to the dev mode startup for large apps.

@aloubyansky any idea?

Reply:
Is it meant to collect Java packages? It's not really doing that. If the idea is to collect the Java packages then the directory that should be analyzed is the classes dir.

Reply:
This is not new code, copied from the old Dev UI, just move from a POST handler to the Processor. Happy to change this, but maybe in a follow up PR ?